WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt

```diff
@@ -93,40 +146,216 @@ class RestoreViewModel @Inject constructor(
+                        activityLogModel = activityLogModel,
+                        toolbarState = DetailsToolbarState(),
+                        items = stateListItemBuilder.buildDetailsListStateItems(
+                                availableItems = availableItems,
+                                published = activityLogModel.published,
+                                onCreateDownloadClick = this@RestoreViewModel::onRestoreSiteClick,
+                                onCheckboxItemClicked = this@RestoreViewModel::onCheckboxItemClicked
+                        ),
+                        type = StateType.DETAILS
+                )
+            } else {
+                transitionToError(GenericFailure)
+            }
         }
     }
 
-    fun addNavigationEventSource(navigationEvent: LiveData<Event<RestoreNavigationEvents>>) {
-        _navigationEvents.addSource(navigationEvent) { event ->
-            _navigationEvents.value = event
+    private fun buildWarning() {
+        _uiState.value = WarningState(
+                toolbarState = WarningToolbarState(),
+                items = stateListItemBuilder.buildWarningListStateItems(
+                        published = restoreState.published as Date,
+                        onConfirmRestoreClick = this@RestoreViewModel::onConfirmRestoreClick,
+                        onCancelClick = this@RestoreViewModel::onCancelClick
+                ),
+                type = StateType.WARNING
+        )
+    }
+
+    private fun buildProgress() {
+        _uiState.value = ProgressState(
+                toolbarState = ProgressToolbarState(),
+                items = stateListItemBuilder.buildProgressListStateItems(
+                        progress = progressStart,
+                        published = restoreState.published as Date,
+                        onNotifyMeClick = this@RestoreViewModel::onNotifyMeClick
+                ),
+                type = StateType.PROGRESS
+        )
+        queryStatus()
+    }
+
+    private fun buildComplete() {
+        _uiState.value = CompleteState(
+                toolbarState = CompleteToolbarState(),
+                items = stateListItemBuilder.buildCompleteListStateItems(
+                        published = restoreState.published as Date,
+                        onDoneClick = this@RestoreViewModel::onDoneClick,
+                        onVisitSiteClick = this@RestoreViewModel::onVisitSiteClick
+                ),
+                type = StateType.COMPLETE
+        )
+    }
+
+    private fun buildError(errorType: RestoreErrorTypes) {
+        _uiState.value = ErrorState(
+                toolbarState = ErrorToolbarState(),
+                items = stateListItemBuilder.buildCompleteListStateErrorItems(
+                        onDoneClick = this@RestoreViewModel::onDoneClick
+                ),
+                errorType = errorType
+        )
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
```

Comment:
🔍 Nitpick: Do we really need the VisibleForTesting annotation? I haven't tested it, but it seems we could mock `wizardManager.navigatorLiveData` in tests? Wdyt?

Reply:
I'll go back and review this, thanks for the direction.